test/jobs/nudge_users_to_request_mentoring_job_test.rb

```diff
@@ -0,0 +1,38 @@
+require "test_helper"
+
+class NudgeUsersToRequestMentoringJobTest < ActiveJob::TestCase
+  test "correct users get solution" do
+    # Set things up for the correct user
+    ruby = create :track, slug: :ruby
+    javascript = create :track, slug: :javascript
+    user = create :user
+    create :iteration, created_at: 2.days.ago, solution: create(:practice_solution, track: javascript, user: user, status: :iterated)
+    create :iteration, created_at: 2.days.ago, solution: create(:practice_solution, track: ruby, user: user, status: :iterated)
+
+    # Too recent
+    create :iteration, created_at: 5.minutes.ago, solution: create(:practice_solution)
+
```

Comment:
Should there be a call to `NudgeUsersToRequestMentoringJob.perform_now` after this to verify that the notification is not sent?

```ruby
User::Notification::Create.expects(:call).never
NudgeUsersToRequestMentoringJob.perform_now
```

If correct, you probably want to do this for the other exception cases below too.

Reply:
Because L35 specifies uses `expects` on `User::Notification::Create`, if only allows calls with the exact params. If that gets called with any of the other users, it explodes. So this is actually tested, but I'll add a comment below to make that clear.

Reply:
Ah okay. Thanks